module/epochs.go

```diff
@@ -36,3 +36,49 @@ type QCContractClient interface {
 	// cluster QC aggregator smart contract for the current epoch.
 	Voted(ctx context.Context) (bool, error)
 }
+
+// DKGContractClient enables interacting with the DKG smart contract. This
+// contract is deployed to the service account as part of a collection of
+// smart contracts that facilitate and manage epoch transitions.
+//
+// Phase parameters may be one of [1,2,3].
+//
+// TODO type for DKGPhase?
+// TODO type for DKGMessage?
+// TODO type for DKGResult? - should be all participant public keys plus group public key
+type DKGContractClient interface {
+
+	// Broadcast broadcasts a message to all other nodes participating in the
+	// DKG for a particular phase. The message is broadcast by submitting a
+	// transaction to the DKG smart contract. An error is returned if the
+	// transaction has failed and should be re-submitted.
+	//
+	// Messages submitted for a phase which has ended will not be accepted and
+	// will result in an error.
+	Broadcast(phase int, msg []byte) error
+
+	// ReadBroadcast reads the broadcast messages from the smart contract for
+	// a particular phase. All messages for the given phase that have been
+	// successfully broadcast will be returned, regardless of whether those
+	// messages have already been read by a previous call to ReadBroadcast.
+	//
+	// Messages are returned in the order in which they were broadcast (received
+	// and stored in the smart contract).
+	//
+	// DKG nodes should call ReadBroadcast one final time once they have
+	// observed the phase deadline trigger to guarantee they receive all
+	// messages for that phase.
+	//
+	// OPTIONAL: add a messageIndex parameter to only receive messages we have
+	// not already received in previous calls
+	ReadBroadcast(phase int) ([][]byte, error)
+
+	// SubmitResult submits the final public result of the DKG protocol. This
+	// represents the node's local computation of the public keys for each
+	// DKG participant and the group public key.
```

Comment:
Just adding that there is an extra `error` output that shows if the DKG was successful. If the error is non-nil, the keys will be nil. This results in the network halting.